lib/Conversion/SeqToSV/SeqToSV.cpp

```diff
@@ -225,15 +318,24 @@ void SeqToSVPass::runOnOperation() {
       memLowering.lowerMemoriesInModule(module, it->second);
   });
 
+  // Mark all ops which can have clock types as illegal.
   ConversionTarget target(*context);
   target.addIllegalDialect<SeqDialect>();
-  target.addLegalDialect<sv::SVDialect, comb::CombDialect, hw::HWDialect>();
+  target.markUnknownOpDynamicallyLegal(isLegalOp);
 
+  SeqToSVTypeConverter typeConverter;
   RewritePatternSet patterns(context);
   patterns.add<CompRegLower<CompRegOp>>(context, lowerToAlwaysFF);
   patterns.add<CompRegLower<CompRegClockEnabledOp>>(context, lowerToAlwaysFF);
+  patterns.add<InstanceLowering>(typeConverter, context);
+  patterns.add<ClockUserLowering<hw::OutputOp>>(typeConverter, context);
+  patterns.add<ClockCastLowering<seq::FromClockOp>>(context);
+  patterns.add<ClockCastLowering<seq::ToClockOp>>(context);
   patterns.add<ClockGateLowering>(context);
   patterns.add<ClockMuxLowering>(context);
+  patterns.add<ClockWireLowering>(context);
+  patterns.add<TypeConversionPattern>(typeConverter, context);
```

Comment:
If the `TypeConversionPattern` deals with `HWModuleOp`s properly, any idea why the `InstanceOp` still needs a dedicated lowering?

Reply:
It actually does. That's great, I removed the patterns.